pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1095,7 +1206,11 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 			newDisk.Driver.IOThread = &ioThreadId
 		}
 
-		domain.Spec.Devices.Disks = append(domain.Spec.Devices.Disks, newDisk)
+		hpStatus, hpOk := c.HotplugVolumes[disk.Name]
+		// if len(c.PermanentVolumes) == 0, it means the vmi is not ready yet, add all disks
+		if _, ok := c.PermanentVolumes[disk.Name]; ok || len(c.PermanentVolumes) == 0 || (hpOk && (hpStatus.Phase == v1.HotplugVolumeMounted || hpStatus.Phase == v1.VolumeReady)) {
+			domain.Spec.Devices.Disks = append(domain.Spec.Devices.Disks, newDisk)
```

Comment:
I *think* this works. but it's scary looking to me that we depend on the length of the PermanentVolumes map being `0` in order to determine that a disk should be added.

How can we improve this? Do we need the PermanentVolumes map at all? Or is there a way we can infer just based if this is a hotplug volume or not?